### PR TITLE
Map infobox styling issue fixes. Resolves #7083

### LINKED ISSUE #7083: More enhancements needed for Infobox
## Description

Some margins are not fully managed in the right way. 

1. In case of text wrapped around an image, the margin is needed for better readability (left and right)

2. In case of titles (eg. H1, H2, H3 etc) the same margin should be applied too. 

3. The header of the Detail panel is smaller than the one used for other panels:

image

For example, for the Catalog tool we have the following:

image

**What kind of improvement you want to add?** (check one with "x", remove the others)
- [x] Minor changes to existing features
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Other useful information
The screenshots above come from [this demo map](https://dev-mapstore.geosolutionsgroup.com/mapstore/#/viewer/openlayers/10358)

### PR DESCRIPTION
**Pull request by alexander-fedorenko**
## Description
Improve styling of the map information modal/sidebar panel.
Adds proper margins to the images so that text wont stick to them from the right side.
Changes left/right margins of the elements to use rem units instead of em to make margins equal for headers and regular text.
Replaces map details sidebar dock with the component that is already in use for catalog and few other plugins.
Updates editor styles to make sure that WYSIWYG editor correctly display margin between floating image and text.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [x] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**
#7083 

**What is the new behavior?**
Proper margins for text heading, images and other elements in the content.
Shared code-base for the map details information container (DockPanel component)

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
image
image
image

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
Seems good overall, please checkout out my comments

Also can you see if you can improve this tiny margin right problem in the preview of the homepage card?
image

File: web/client/components/details/DetailsPanel.jsx [hunk 1]
```diff
@@ -36,53 +31,34 @@ class DetailsPanel extends React.Component {
...
-            </ResizeDetector>
+            <DockPanel
+                open={this.props.active}
+                size={660}
+                position="right"
+                bsStyle="primary"
+                title={<Message msgId="details.title"/>}
+                onClose={() => this.props.onClose()}
+                glyph="sheet"
+                style={{height: 'calc(100% - 30px)'}}
```
[the remark this answers is not in the document]
Comment (alexander-fedorenko) [reply web/client/components/details/DetailsPanel.jsx]: Got back selector value, it was available via dockStyle property, but I removed it eventually after checking MapCatalog plugin as a reference.
Also resolved margin issue that appears on preview of the details.

**Review by MV88 — APPROVED** [review]
LGTM

**Comment by MV88** [comment]
@ElenaGallo please test in DEV

**Comment by ElenaGallo** [comment]
Test passed, @alexander-fedorenko please backport to 2021.02.xx. Thanks

**Pull request merged**